src/lib/storage/zero_suppression/simd_bp128/simd_bp128_vector.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "storage/zero_suppression/base_zero_suppression_vector.hpp"
+
+#include "oversized_types.hpp"
+#include "simd_bp128_decoder.hpp"
+#include "simd_bp128_iterator.hpp"
+
+#include "types.hpp"
+
+namespace opossum {
+
+/**
+ * @brief Bit-packed vector with varying bit length
+ *
+ * Values are compressed in blocks of 128 integers. Each block has its own bit-length.
+ * Sixteen blocks combine to form a meta block of 2048 values. Bit-length information
+ * are stored per meta block in 128 bit (8 bit for each bit length).
```

Comment:
The documentation for this seems to be scattered across several files. Also, an ASCII art of the actual layout would help a lot.

Reply:
If someone really wants to dive into the code, he or she should read the paper which is referenced in `SimdBp128Packing`. To get a simple understanding, this should (might) help. The comments for `SimdBp128Packing` explain specific implementation details, which are not described in the paper and which differ from the reference implementation.